src/nxt_h1proto.c

```diff
@@ -2353,6 +2362,61 @@ nxt_h1p_peer_header_send(nxt_task_t *task, nxt_http_peer_t *peer)
     }
 
     nxt_conn_write(task->thread->engine, c);
+
+    return;
+
+fail:
+
+    r->state->error_handler(task, r, peer);
+}
+
+
+static nxt_int_t
+nxt_h1p_peer_request_target(nxt_http_request_t *r, nxt_str_t *target)
+{
+    u_char  *p;
+    size_t  size, encode;
+
+    if (!r->uri_changed) {
+        *target = r->target;
+        return NXT_OK;
+    }
+
+    if (!r->quoted_target && r->args->length == 0) {
+        *target = *r->path;
+        return NXT_OK;
+    }
+
+    if (r->quoted_target) {
+        encode = nxt_encode_complex_uri(NULL, r->path->start,
+                                        r->path->length);
+    } else {
+        encode = 0;
+    }
+
+    size = r->path->length + encode * 2 + 1 + r->args->length;
+
+    target->start = nxt_mp_nget(r->mem_pool, size);
+    if (target->start == NULL) {
+        return NXT_ERROR;
+    }
+
+    if (r->quoted_target) {
+        p = (u_char *) nxt_encode_complex_uri(target->start, r->path->start,
+                                              r->path->length);
```

Comment:
This begged the question, will a `size_t` always be big enough to hold a `uintptr_t`.

While I don't believe the C standard makes any guarantees that that will be the case, I think in practice it will be, at least for any system we care about.

On 64bit Linux (and macOS) they are defined like

```
$ echo | gcc -E -xc -include 'stddef.h' - | grep size_t
typedef long unsigned int size_t;
$ echo | gcc -E -xc -include 'stdint.h' - | grep uintptr_t
typedef unsigned long int uintptr_t;
```

On 32bit Linux

```
$ echo | gcc -E -xc -include 'stddef.h' - | grep size_t
typedef unsigned int size_t;
$ echo | gcc -E -xc -include 'stdint.h' - | grep uintptr_t
typedef unsigned int uintptr_t;
```

Reply:
I replaced it with `nxt_uint_t` that referred to `nxt_http_return.c`.
I feel both `nxt_uint_t` and `size_t` are correct since the max size is limited by the `large_header_buffer_size` option that is validated by the Unit JSON. And the max number length is 14.
```
#define NXT_CONF_MAX_NUMBER_LEN    14
```

Reply:
That makes me more nervous! What if that constant ever increases... I think it was **safer** with `size_t`.

Reply:
I kept `size_t`.